controllers/workspace/provision/routing.go

```diff
@@ -122,13 +122,20 @@ func getSpecRouting(
 		}
 	}
 
+	var annotations map[string]string
+	if val, ok := workspace.Annotations[config.WorkspaceRestrictedAccessAnnotation]; ok {
+		annotations = map[string]string{}
+		annotations[config.WorkspaceRestrictedAccessAnnotation] = val
+	}
+
 	routing := &v1alpha1.WorkspaceRouting{
 		ObjectMeta: v1.ObjectMeta{
 			Name:      fmt.Sprintf("routing-%s", workspace.Status.WorkspaceId),
 			Namespace: workspace.Namespace,
 			Labels: map[string]string{
 				config.WorkspaceIDLabel: workspace.Status.WorkspaceId,
 			},
+			Annotations: annotations,
```

Comment:
Seems since we set annotations, we need to adapt routingDiffOpts, otherwise, resricted-access annotation changes are not propagated from DevWorkspace to its routing.

Reply:
Good catch! Will update.

Reply:
I put a bit of work into this but it's pretty ugly as is. Another option might be to use `cmp.FilterPath` but I couldn't get it to work as expected.

Reply:
It was just an hour of tries and failures and seems I made it work, I still don't understand how it's going to work.

https://github.com/devfile/devworkspace-operator/commit/456d09a5ea151ee2bd4f406d600f8fa7e383f473

How to test:
```bash
make run
#1
cat <<EOF | kubectl apply -f -
kind: DevWorkspace
apiVersion: workspace.devfile.io/v1alpha2
metadata:
  name: theia
spec:
  started: true
  routingClass: non-supported-yet
  template:
    projects:
      - name: project
        git:
          remotes:
            origin: "https://github.com/che-samples/web-nodejs-sample.git"
    components:
      - name: theia
        plugin:
          id: eclipse/che-theia/next
      - name: terminal
        plugin:
          id: eclipse/che-machine-exec-plugin/nightly
    commands:
      - id: say hello
        exec:
          component: plugin
          commandLine: echo "Hello from $(pwd)"
          workingDir: ${PROJECTS_ROOT}/project/app
EOF
#2
DW_ID=$(kubectl get dw theia -o=yaml | yq '.status.workspaceId' -r)

#3
kubectl patch workspaceroutings.controller.devfile.io routing-$DW_ID --type=merge --patch '
metadata:
  finalizers: ["my-custom"]
'
# ^ operator logs contains only  We are not interested in made routing update

#4
kubectl patch workspaceroutings.controller.devfile.io routing-$DW_ID --type=merge --patch '
metadata:
  labels:
    any: any
'
# ^ operator logs contains:
#  We are not interested in made routing update 3x or even more +
#  We are interested in made routing update 1x +
```

Reply:
This is similar to what I had when I tried, but I was having trouble verifying it worked exactly as I expected. I think using filterPath is kind of harder to read/parse, even though the huge list of fields is ugly.

Eventually I'd like to move away from the `cmp` package -- it's listed as explicitly not intended for production and we've been using it as a crutch. In working on https://github.com/devfile/devworkspace-operator/pull/229 I came across the k8s `equality` package, which looks like it'll do what we want in a cleaner way and I'd like to work on migrating to that sometime soon.

Reply:
> it's listed as explicitly not intended for production and we've been using it as a crutch

Good catch!
Before I found 
> Its propensity towards panicking means that its unsuitable for production environments where a spurious panic may be fatal.

I played and came solution where we have not bad readability https://github.com/devfile/devworkspace-operator/commit/20e92d3d3be6efa276eb95cd7bbec20d7ae68822
But it does not matter because it's not production-ready ) One day we can use it for tests only :-D

So, I'm OK with `equality` I just don't understand in details how the same would look like there.